test/python/circuit/test_circuit_equivalence_checker.py

```diff
@@ -0,0 +1,126 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+
+"""Test Qiskit's Circuit Equivalence Checker"""
+
+import unittest
+
+from qiskit.test import QiskitTestCase
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.tools import UnitaryEquivalenceChecker
+
+
+class TestEquivalenceChecker(QiskitTestCase):
+    """Test equivalence checker"""
+
+    def setUp(self) -> None:
+        super().setUp()
+        try:
+            from qiskit import Aer   # pylint: disable=unused-import
+            self.aer_installed = True
+        except ImportError:
+            self.aer_installed = False
+
+    def verify_result(self, checkers, circ1, circ2,
+                      success, equivalent,
+                      phase):
+        """
+        Run the checkers, and verify that their outputs
+        match the expected result
+        """
+
+        for checker in checkers:
```

Comment:
use the decorator to create subtests (see how it is done in other qiskit tests)

Reply:
done